CRM/Upgrade/Incremental/php/FiveThirtySeven.php

```diff
@@ -106,4 +107,42 @@ public static function alterSavedSearchFK(CRM_Queue_TaskContext $ctx) {
     return TRUE;
   }
 
+  /**
+   * Install recaptcha extension.
+   *
+   * This feature is restructured as a core extension - which will eventually allow us to replace/remove the
+   * reCAPTCHA implementation
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public static function installReCaptchaExtension(CRM_Queue_TaskContext $ctx) {
+    // Install via direct SQL manipulation. Note that:
+    // (1) This extension has no activation logic.
+    // (2) On new installs, the extension is activated purely via default SQL INSERT.
+    // (3) Caches are flushed at the end of the upgrade.
+    // ($) Over long term, upgrade steps are more reliable in SQL. API/BAO sometimes don't work mid-upgrade.
+    $insert = CRM_Utils_SQL_Insert::into('civicrm_extension')->row([
+      'type' => 'module',
+      'full_name' => 'recaptcha',
+      'name' => 'reCAPTCHA',
+      'label' => 'reCAPTCHA',
+      'file' => 'recaptcha',
+      'schema_version' => NULL,
+      'is_active' => 1,
+    ]);
+    CRM_Core_DAO::executeQuery($insert->usingReplace()->toSQL());
+
+    CRM_Core_DAO::executeQuery('
+UPDATE civicrm_navigation
+SET name="Misc (Undelete, PDFs, Limits, Logging, etc.)",label="Misc (Undelete, PDFs, Limits, Logging, etc.)"
+WHERE name="Misc (Undelete, PDFs, Limits, Logging, Captcha, etc.)"
+    ');
```

Comment:
Nice and thorough :)

May I suggest splitting this into a couple statements. There is a a slim possibility that the `label` has been customized, and this would preserve the customization.

```suggestion
CRM_Core_DAO::executeQuery('
UPDATE civicrm_navigation
SET name="Misc (Undelete, PDFs, Limits, Logging, etc.)"
WHERE name="Misc (Undelete, PDFs, Limits, Logging, Captcha, etc.)"
    ');
CRM_Core_DAO::executeQuery('
UPDATE civicrm_navigation
SET label="Misc (Undelete, PDFs, Limits, Logging, etc.)"
WHERE label="Misc (Undelete, PDFs, Limits, Logging, Captcha, etc.)"
    ');
```

Note that I skimmed an `fr_FR` build and a multilingual build, and I think this will be OK.  It seems that `civicrm_navigation`.`label` is ordinarily stored as English and translated at runtime. So the only deviation I could reproduce required editing `Administer => Customize => Navigation`.

Reply:
Did an `r-run` of the upgrader - confirmed that either formulation works on a stock configuration.

(I'd still offer the suggestion for customized configurations.)